crates/subspace-networking/src/request_handlers/piece_announcement.rs

```diff
@@ -0,0 +1,52 @@
+//! Piece announcement request response protocol..
+//!
+//! Handle (i.e. answer) pieces announcement requests from a remote peer received via
+//! `RequestResponsesBehaviour` with generic [`GenericRequestHandler`].
+
+use crate::request_handlers::generic_request_handler::{GenericRequest, GenericRequestHandler};
+use libp2p::Multiaddr;
+use parity_scale_codec::{Decode, Encode};
+use tracing::debug;
+
+/// Piece announcement protocol request.
+#[derive(Debug, Clone, Eq, PartialEq, Encode, Decode)]
+pub struct PieceAnnouncementRequest {
+    /// Request key - piece index multihash
+    pub piece_key: Vec<u8>,
+
+    /// External addresses of the peer
+    pub addresses: Vec<Vec<u8>>,
```

Comment:
Why are they in raw bytes? Looks like having `Multiaddr` here would make much more sense and make somewhat confusing `converted_addresses` unnecessary.

Reply:
That only means you can't derive implementation, but you can still implement it manually, `Multiaddr` is not hard to convert to bytes and back. You are doing it anyway, the only difference is that instead of doing it in one place and work with mostly stack-allocated data you're doing it in every place where `PieceAnnouncementRequest` is used and have to do multiple allocations in `Vec<Vec<u8>>`.